### PR TITLE
Make internal prefixes available to linter rules

### PR DESCRIPTION
**Pull request by jabraham17**
Makes the internal prefixes (as specified by `--internal-prefix chpl_`) be exposed to linter rules, rather than just the driver. Technically this was already possible, this PR just makes the naming based rules exclude the prefixes, such that `chpl_camelCase` is considered to be camelCase

[Reviewed by @DanilaFe]

### CODE REVIEW

File: tools/chplcheck/src/rules.py [hunk 1]
```diff
@@ -147,8 +159,11 @@ def CamelOrPascalCaseVariables(context: Context, node: VarLikeDecl):
             return True
         if node.linkage() == "extern":
             return True
-        return check_camel_case(context, node) or check_pascal_case(
-            context, node
+        internal_prefixes = driver.config.internal_prefixes
+        return check_camel_case(
+            context, node, internal_prefixes=internal_prefixes
+        ) or check_pascal_case(
+            context, node, internal_prefixes=internal_prefixes
```
Comment (DanilaFe) [inline tools/chplcheck/src/rules.py:166]: kwarg form is unnecessary here and very verbose IMO

Comment (DanilaFe) [reply tools/chplcheck/src/rules.py:166]: Ditto in all of these, actually.

**Review by DanilaFe — APPROVED** [review]
Looks good, but needs tests for the new flag.

**Comment by jabraham17** [comment]
I removed the kwargs.

How do you feel about waiting for https://github.com/chapel-lang/chapel/pull/26191 to add the tests? That will add `.chplcheckopts` to the chplcheck test infrastructure and will make adding tests for this much easier

**Comment by DanilaFe** [comment]
Sounds good to me.

**Pull request merged**